rust/agama-cli/src/profile.rs

```diff
@@ -45,15 +56,55 @@ fn validate(path: String) -> anyhow::Result<()> {
 fn evaluate(path: String) -> anyhow::Result<()> {
     let evaluator = ProfileEvaluator {};
     evaluator
-        .evaluate(Path::new(&path))
+        .evaluate(Path::new(&path), stdout())
         .context("Could not evaluate the profile".to_string())?;
     Ok(())
 }
 
-pub fn run(subcommand: ProfileCommands) -> anyhow::Result<()> {
+async fn import(url: String, dir: Option<PathBuf>) -> anyhow::Result<()> {
+    let tmpdir = TempDir::new()?; // TODO: create it only if dir is not passed
+    let output_file = if url.ends_with(".sh") {
+        "profile.sh"
+    } else if url.ends_with(".jsonnet"){
+        "profile.jsonnet"
+    } else {
+        "profile.json"
+    };
+    let output_dir = dir.unwrap_or_else(|| tmpdir.into_path());
+    let reader = ProfileReader::new(url.as_str())?;
+    let contents = reader.read()?;
+    let mut output_path = output_dir.join(output_file);
+    let mut output_fd = File::create(output_path.clone())?;
+    output_fd.write_all(contents.as_bytes())?;
+    if output_file.ends_with(".sh") {
+        let err = Command::new("bash")
+        .args([output_path.to_str().context("Wrong path to shell script")?])
+        .exec();
+        println!("Error: {}", err);
+    }
+    
+    if output_file.ends_with(".jsonnet") {
+        let fd = File::create(output_dir.join("profile.json"))?;
+        let evaluator = ProfileEvaluator {};
+        evaluator
+            .evaluate(&output_path, fd)
+            .context("Could not evaluate the profile".to_string())?;
+        output_path = output_dir.join("profile.json");
+    }
+
+    let output_path_str = output_path.to_str().context("Failed to get output path")?.to_string();
+    // TODO: optional skip of validation
+    validate(output_path_str.clone())?;
+    crate::config::run(crate::config::ConfigCommands::Load { path: output_path_str }, crate::printers::Format::Json).await?;
```

Comment:
Oh, it looks like we're calling the same binary again with different options. I'm not sure I like this approach.

Perhaps we might think about building a `Config` API that can be used in `config::run`. Let's keep it by now but, please, add a FIXME.

Reply:
in fact it is not same binary...I just call method that implements `config load`. Otherwise I will call just `agama config load` via Command, but I think this will be faster?

Reply:
yeah, reason is that config load has it inside this run command - https://github.com/openSUSE/agama/blob/master/rust/agama-cli/src/config.rs#L89

Reply:
Yes, I know the reason :-) and I dislike it. I will add this to the list of things to fix in the Rust codebase.